src/main/webapp/app/shared/participant-scores/participant-scores.service.ts

```diff
@@ -21,6 +21,17 @@ export class ParticipantScoreDTO {
     public lastRatedResultScore: number | null;
 }
 
+/**
+ * Normally the use of null in the client is discouraged but in this case it is fine as the server will always send all
+ * the properties but will give them the value null (a property will never be undefined here).
+ */
+export class ParticipantScoreAverageDTO {
+    public userName: string | null;
+    public teamName: string | null;
+    public averageScore: number | null;
+    public averageRatedScore: number | null;
```

Comment:
are you sure you need `null` instead of `undefined`?

Reply:
Yes. In this case here yes. I also wrote the following comment:
`/**
 * Normally the use of null in the client is discouraged but in this case it is fine as the server will always send all
 * the properties but will give them the value null (a property will never be undefined here).
 */
export class ParticipantScoreDTO {
    public id: number | null;
    public userId: number | null;
    public userName: string | null;
    public teamId: number | null;
    public teamName: string | null;
    public exerciseId: number | null;
    public exerciseTitle: string | null;
    public lastResultId: number | null;
    public lastResultScore: number | null;
    public lastRatedResultId: number | null;
    public lastRatedResultScore: number | null;
}`